consensus/src/pos/error.rs

```diff
@@ -30,6 +30,8 @@ pub enum ConsensusPoSError {
     PropertyQueryError(#[from] chainstate_types::PropertyQueryError),
     #[error("Stake kernel hash failed to meet the target requirement")]
     StakeKernelHashTooHigh,
+    #[error("Epoch data not provided")]
+    NoEpochData,
```

Comment:
provide epoch index

Reply:
This error here means that epoch data wasn't provided by the caller of the function (as it's an `Option`), and not that epoch data could not be found, so I don't think a height or index here will help.

Reply:
The reason why it's optional is because it's passed to the general `generate_finalize_block_data()` function, and then only looks at it epoch data if we're wanting PoS.

It's as if we need two types of parameter enums - one for user input over the wire, and another for internal data. This can be done, but I think that may be a bigger change, which we can probably put off for another PR?

Reply:
ok